x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/io/stream/PlanStreamInput.java

```diff
@@ -52,7 +70,7 @@ public PlanStreamInput(
         NamedWriteableRegistry namedWriteableRegistry,
         EsqlConfiguration configuration
     ) {
-        this(streamInput, registry, namedWriteableRegistry, configuration, DEFAULT_NAME_ID_FUNC);
+        this(streamInput, registry, namedWriteableRegistry, configuration, DEFAULT_NAME_ID_FUNC.get());
```

Comment:
Should we remove the other constructor to disallow passing the nameIdFunction parameter. I think we also need to update the comment for the nameIdFunction field. It's outdated.